### PR TITLE
Use cobra

### PR DESCRIPTION
**Pull request by ptolemy428**
## Approvals

- [x] Verified
- [ ] @TimMoore
- [ ] @hebbo
- [x] @ptolemy428
- [ ] @iiman

### CODE REVIEW

**Comment by ptolemy428** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;QdZyjrn9&#x2F;371-enhance-aws-nquire-to-get-the-latest-ami-by-name-and-branch"> Enhance aws-nquire to get the latest ami by name and branch</a>

File: command/ami/command.go [hunk 1]
```diff
@@ -35,35 +31,62 @@ func Run(prefix string, field string, region string) string {
...
 	}
-	return resp
 }
 
-func params() *ec2.DescribeImagesInput {
-	return &ec2.DescribeImagesInput{
-		Owners: []*string{
-			aws.String("self"),
-		},
+func filterByBranchTag(i *ec2.Image, key string, value string) bool {
```
Comment (TimMoore) [inline command/ami/command.go]: Could this be done by passing another filter to the `DescribeImages` API?

Comment (ptolemy428) [inline command/ami/command.go]: @TimMoore, this is a very good question. I was toggling between the decisions of server side v.s client side filtering a bit. The main reason is that aws doco on the behaviour of using tags to search ami is not clear on the exact behaviour of accepting tags. It's actually a bit ambiguous.
http://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html

I switched to server side filtering of ami name in this PR, which seems to be working fine so far. I was hoping to test it a bit more on the CI before fully committing to it. Client side filtering on the tag branch is less performant than server side, but gives us full control of logics.. I was planning to optimise if the testing goes well.

Comment (TimMoore) [inline command/ami/command.go]: OK sounds good

**Pull request merged**